packages/core/src/internal/utils/resolve-module-parameter.ts

```diff
@@ -13,35 +13,28 @@ export function resolveModuleParameter(
   moduleParamRuntimeValue: ModuleParameterRuntimeValue<ModuleParameterType>,
   context: { deploymentParameters: DeploymentParameters; accounts: string[] }
 ): SolidityParameterType {
-  if (context.deploymentParameters === undefined) {
-    assertIgnitionInvariant(
-      moduleParamRuntimeValue.defaultValue !== undefined,
-      `No default value provided for module parameter ${moduleParamRuntimeValue.moduleId}/${moduleParamRuntimeValue.name}`
-    );
+  const potentialParamAtModuleLevel =
```

Comment:
I have switched the logic here.

Instead of looking up the module parameters and falling back to the global parameters then check if the named parameter exists within the object, we change to:

* look up the named parameter in the module level params
* look up the named parameter in the global level params
* finally use the default value